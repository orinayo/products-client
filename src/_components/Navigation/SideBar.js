import React from 'react'
import Icon from '../Icon/Icon'
import { Link } from 'react-router-dom'

const SideBar = () => {
  return (
    <nav className='sidebar'>
      <ul className='side-nav'>
        <li className={`side-nav__item`}>
          <Link to='/products/new' className='side-nav__link'>
            <Icon name='plus' />
            <span>Add Product</span>
          </Link>
        </li>
        <li className='side-nav__item'>
          <a href='www.google.com' target='_blank' className='side-nav__link'>
            <Icon name='open-book' />
            <span>View API Docs</span>
          </a>
        </li>
      </ul>
      <div className='legal'>
        &copy; 2018
      </div>
    </nav>
  )
}

export default SideBar
