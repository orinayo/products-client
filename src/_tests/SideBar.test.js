import { shallow } from 'enzyme'
import React from 'react'
import SideBar from '../_components/Navigation/SideBar'

it('expect to render SideBar component', () => {
  expect(shallow(<SideBar />)).toMatchSnapshot()
})