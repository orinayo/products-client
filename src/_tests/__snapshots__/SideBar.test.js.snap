// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render SideBar component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul
          className="side-nav"
        >
          <li
            className="side-nav__item"
          >
            <Link
              className="side-nav__link"
              replace={false}
              to="/products/new"
            >
              <Icon
                name="plus"
              />
              <span>
                Add Product
              </span>
            </Link>
          </li>
          <li
            className="side-nav__item"
          >
            <a
              className="side-nav__link"
              href="www.google.com"
              target="_blank"
            >
              <Icon
                name="open-book"
              />
              <span>
                View API Docs
              </span>
            </a>
          </li>
        </ul>,
        <div
          className="legal"
        >
          © 2018
        </div>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="side-nav__item"
            >
              <Link
                className="side-nav__link"
                replace={false}
                to="/products/new"
              >
                <Icon
                  name="plus"
                />
                <span>
                  Add Product
                </span>
              </Link>
            </li>,
            <li
              className="side-nav__item"
            >
              <a
                className="side-nav__link"
                href="www.google.com"
                target="_blank"
              >
                <Icon
                  name="open-book"
                />
                <span>
                  View API Docs
                </span>
              </a>
            </li>,
          ],
          "className": "side-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="side-nav__link"
                replace={false}
                to="/products/new"
              >
                <Icon
                  name="plus"
                />
                <span>
                  Add Product
                </span>
              </Link>,
              "className": "side-nav__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    name="plus"
                  />,
                  <span>
                    Add Product
                  </span>,
                ],
                "className": "side-nav__link",
                "replace": false,
                "to": "/products/new",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "plus",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Product",
                  },
                  "ref": null,
                  "rendered": "Add Product",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="side-nav__link"
                href="www.google.com"
                target="_blank"
              >
                <Icon
                  name="open-book"
                />
                <span>
                  View API Docs
                </span>
              </a>,
              "className": "side-nav__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Icon
                    name="open-book"
                  />,
                  <span>
                    View API Docs
                  </span>,
                ],
                "className": "side-nav__link",
                "href": "www.google.com",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "open-book",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "View API Docs",
                  },
                  "ref": null,
                  "rendered": "View API Docs",
                  "type": "span",
                },
              ],
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© 2018",
          "className": "legal",
        },
        "ref": null,
        "rendered": "© 2018",
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="side-nav"
          >
            <li
              className="side-nav__item"
            >
              <Link
                className="side-nav__link"
                replace={false}
                to="/products/new"
              >
                <Icon
                  name="plus"
                />
                <span>
                  Add Product
                </span>
              </Link>
            </li>
            <li
              className="side-nav__item"
            >
              <a
                className="side-nav__link"
                href="www.google.com"
                target="_blank"
              >
                <Icon
                  name="open-book"
                />
                <span>
                  View API Docs
                </span>
              </a>
            </li>
          </ul>,
          <div
            className="legal"
          >
            © 2018
          </div>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="side-nav__item"
              >
                <Link
                  className="side-nav__link"
                  replace={false}
                  to="/products/new"
                >
                  <Icon
                    name="plus"
                  />
                  <span>
                    Add Product
                  </span>
                </Link>
              </li>,
              <li
                className="side-nav__item"
              >
                <a
                  className="side-nav__link"
                  href="www.google.com"
                  target="_blank"
                >
                  <Icon
                    name="open-book"
                  />
                  <span>
                    View API Docs
                  </span>
                </a>
              </li>,
            ],
            "className": "side-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="side-nav__link"
                  replace={false}
                  to="/products/new"
                >
                  <Icon
                    name="plus"
                  />
                  <span>
                    Add Product
                  </span>
                </Link>,
                "className": "side-nav__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Icon
                      name="plus"
                    />,
                    <span>
                      Add Product
                    </span>,
                  ],
                  "className": "side-nav__link",
                  "replace": false,
                  "to": "/products/new",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "plus",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Add Product",
                    },
                    "ref": null,
                    "rendered": "Add Product",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="side-nav__link"
                  href="www.google.com"
                  target="_blank"
                >
                  <Icon
                    name="open-book"
                  />
                  <span>
                    View API Docs
                  </span>
                </a>,
                "className": "side-nav__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Icon
                      name="open-book"
                    />,
                    <span>
                      View API Docs
                    </span>,
                  ],
                  "className": "side-nav__link",
                  "href": "www.google.com",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "open-book",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "View API Docs",
                    },
                    "ref": null,
                    "rendered": "View API Docs",
                    "type": "span",
                  },
                ],
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "© 2018",
            "className": "legal",
          },
          "ref": null,
          "rendered": "© 2018",
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
